pkg/search/postgres/query/common.go

```diff
@@ -91,5 +95,26 @@ func MatchFieldQuery(dbField *walker.Field, value string, highlight bool, now ti
 	if !ok {
 		return nil, nil
 	}
-	return matchFieldQuery(dbField, field, value, highlight, now)
+
+	qualifiedColName := dbField.Schema.Table + "." + dbField.ColumnName
+	dataType := dbField.DataType
+	var goesIntoHavingClause bool
+	if derivedMetadata != nil {
+		switch derivedMetadata.DerivationType {
+		case search.CountDerivationType:
+			qualifiedColName = fmt.Sprintf("count(%s)", qualifiedColName)
+			dataType = walker.Integer
```

Comment:
make `dataType` property of CountDerivationType?

Reply:
No, I think it belongs here. This is SQL implementation detail-specific, while the fields in `search/options.go` are intended to be descriptive metadata that don't have any implementation information.